File: src/main/java/io/srmppn/HappyDorm/room/api/RoomReservation.java
package io.srmppn.HappyDorm.room.api;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

import java.util.UUID;

public class RoomReservation {
    public static class ReserveRoomCommand {
        @TargetAggregateIdentifier
        private UUID roomId;
        private UUID accountId;

        public ReserveRoomCommand(UUID roomId, UUID accountId) {
            this.roomId = roomId;
            this.accountId = accountId;
        }

        public UUID getRoomId() {
            return roomId;
        }

        public UUID getAccountId() {
            return accountId;
        }
    }

    public static class RoomReservedEvent {
        private UUID roomId;
        private UUID accountId;

        public RoomReservedEvent(UUID roomId, UUID accountId) {
            this.roomId = roomId;
            this.accountId = accountId;
        }

        public UUID getRoomId() {
            return roomId;
        }

        public void setRoomId(UUID roomId) {
            this.roomId = roomId;
        }

        public UUID getAccountId() {
            return accountId;
        }
    }

    public static class UnreserveRoomCommand {
        @TargetAggregateIdentifier
        private UUID roomId;

        public UnreserveRoomCommand(UUID roomId) {
            this.roomId = roomId;
        }

        public UUID getRoomId() {
            return roomId;
        }
    }

    public static class RoomUnreservedEvent {
        private UUID roomId;

        public RoomUnreservedEvent(UUID roomId) {
            this.roomId = roomId;
        }

        public UUID getRoomId() {
            return roomId;
        }

        public void setRoomId(UUID roomId) {
            this.roomId = roomId;
        }
    }
}

File: src/main/java/io/srmppn/happydorm/room/api/room-reservation.kt
package io.srmppn.happydorm.room.api

import org.axonframework.modelling.command.TargetAggregateIdentifier
import org.axonframework.serialization.Revision
import java.util.*

data class ReserveRoomCommand(
  @TargetAggregateIdentifier
  val roomId: UUID,
  val accountId: UUID
)

@Revision("1.0")
data class RoomReservedEvent(
  val roomId: UUID,
  val accountId: UUID
)

class UnreserveRoomCommand(
  @TargetAggregateIdentifier
  val roomId: UUID
)

@Revision("1.0")
class RoomUnreservedEvent(
  val roomId: UUID
)

@Revision("1.0")
class RoomReservationExpiredEvent(
  val roomId: UUID
)

@Revision("1.0")
data class RoomRentedEvent(
  val roomId: UUID
)